### ISSUE TITLE
Roslyn doesn't recognize properties' accesses in ??= assignments when checking initialization of non-nullable members

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#44212]
**Version Used**: 

```
commit 7b27ca3b7722c5e56910a6cfba837e6a9d352d4f
Author: Tom<C3><A1><C5><A1> Matou<C5><A1>ek <tmat@users.noreply.github.com>
Date:   Mon May 11 19:17:04 2020 -0700

    Fixes PathMap parsing to support all paths (#43706)
```

**Steps to Reproduce**:

Compile the following code
```cs
using System.Diagnostics.CodeAnalysis;
#nullable enable
class C
{
  string f;
  
  public C() // CS8618: Non-nullable field 'f' is uninitialized. Consider declaring the field as nullable.
  {
    Prop ??= "";
    f.ToString(); // no warnings
  }
  
  public C(byte b) // no warnings
  {
    if (Prop == null)
    {
      Prop = "";
    }
    f.ToString(); // no warnings
  }
  
  [MemberNotNull(nameof(f))]
  string? Prop
  {
    get => f = "";
    set => f = value ?? "";
  }
}
```

**Expected Behavior**:
No warnings in both constructors. Their code is equivalent.

**Actual Behavior**:
` CS8618: Non-nullable field 'f' is uninitialized. Consider declaring the field as nullable.` is reported for the first constructor only.
